transmission_interface/src/transmission_loader.cpp

```diff
@@ -62,12 +62,12 @@ TransmissionLoader::getActuatorReduction(const TiXmlElement& parent_el,
   {
     ROS_ERROR_STREAM_NAMED("parser", "Actuator '" << actuator_name << "' of transmission '" << transmission_name <<
                            "' specifies the <mechanicalReduction> element, but is not a number.");
-    return ParseStatus::BAD_TYPE;
+    return false;
```

Comment:
I liked @ipa-mdl's suggestion in #333 of using exceptions for this case, to differentiate from the "element not found" case. Or perhaps the best solution would be:

 - An exception for "Required element not found"
 - An exception for "Element is not a number/BAD_TYPE"
 - False for "Optional element not found"
 - True for success

I'm just hesitant to lump together the error conditions.

Reply:
I'm not sure I fully agree with returning false if an optional element wasn't found. In that case the caller would need to know more information otherwise things would likely terminate even though the only thing that happened was an optional thing wasn't found.

Reply:
> the caller would need to know more information

I would argue that with the exception model I mentioned, you *would* be giving them sufficient information, since it differentiates between exceptions/an error vs false/"optional not found". Plus this is an internal API anyways

The main reason I'm pushing this is because "Optional element not found" used to return `ParseStatus::NO_DATA`, so it makes me a bit uncomfortable to change it to a silent failure.

Reply:
yeah I think my main point of contention is that in the case of `ParseStatus::NO_DATA` it was already a "silent" failure in which you would only get a rosconsole message. Just looking at one of the callers to these functions [here](https://github.com/ros-controls/ros_control/blob/noetic-devel/transmission_interface/src/differential_transmission_loader.cpp#L234). It only actually checks if the return type is `ParseStatus::BAD_TYPE`, so if it's `ParseStatus::SUCCESS` or `ParseStatus::NO_DATA` it will simply continue on.

Reply:
Ah I see, that's fair. I hadn't noticed that every time it's called with optional, the existing callers makes a point to only check `BAD_TYPE` rather than `!SUCCESS`.

Reply:
Let me know if there is some action you'd like me to take here, otherwise I think I addressed the other comment. Thanks!

Reply:
LGTM